packaging/hudi-hadoop-mr-bundle/pom.xml

```diff
@@ -77,6 +79,10 @@
                   <pattern>com.esotericsoftware.minlog.</pattern>
                   <shadedPattern>org.apache.hudi.com.esotericsoftware.minlog.</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>org.apache.avro.</pattern>
+                  <shadedPattern>org.apache.hudi.org.apache.avro.</shadedPattern>
```

Comment:
Add one more property and use it here.
mr.bundle.avro.shade.prefix
  <shadedPattern>${mr.bundle.avro.shade.prefix}org.apache.avro.</shadedPattern>

The default value should be empty string "" 
<mr.bundle.avro.shade.prefix></mr.bundle.avro.shade.prefix>

and in the profile activation step (shade-avro), set it to 
<mr.bundle.avro.shade.prefix>org.apache.hudi.</mr.bundle.avro.shade.prefix>


Can you try this setup and check.